dream_layer_backend/shared_utils.py

```
@@ -286,4 +287,211 @@ def upload_controlnet_image(file, unit_index: int = 0) -> Dict[str, Any]:
+
+        # Check if PromptServer instance exists
+        if hasattr(PromptServer, 'instance') and PromptServer.instance is not None:
+            # Create the WebSocket event data
+            event_data = {
+                "model_type": model_type,
+                "filename": filename,
+                "action": "added",
+                "timestamp": int(time.time() * 1000)
+            }
+
+            print(f"📡 Emitting WebSocket event: models-refresh")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)